### ISSUE TITLE
Handle asynchronous messages

### ISSUE THREAD

**Issue by DenisPeskov** [ALLAN-DIP/baseline_bots#145]
Currently messages get lost due to time out, which causes conversations to be incomplete even between two versions of our bot.

Solution will allow a conversation to continue as long as moves are being proposed.

**Comment by aphedges** [comment]
I've been investigating this over the past couple of weeks, and I finally reached a point where I was able to verify the bug was happening as described and to determine what is causing the "bug".

It appears that [`SmartOrderAccepterBot`](https://github.com/ALLAN-DIP/baseline_bots/blob/a555cde46037fc5b9c6861ba61bacbdc08886b39/src/baseline_bots/bots/smart_order_accepter_bot.py) only reads messages at the start of each phase. Although it sends messages throughout the phase, there is no "message loop" to ensure that back-and-forth communication happens. In addition, it does not appear that messages from previous phases are taken into account, either. Changing this will require some larger architectural changes. I created #167 as some preliminary code cleanup to make that easier, but a lot more work will need to be done.
